public/talos/v1.11/security/talos-security-checklist.mdx

```diff
@@ -0,0 +1,160 @@
+---
+title: Talos Security Checklist
+description: A practical checklist for securing Talos Linux clusters.
+---
+
+Talos is designed to be secure by default, no SSH, no interactive shell, an API-driven OS, and an immutable filesystem.
+
+Still, there are several additional steps you can take to strengthen the security posture of your cluster.
+
+This checklist highlights the key areas to review and configure when running Talos in production environments.
+
+## Protect secrets and sensitive information
+
+Talos stores certificates, keys, and other sensitive configuration data inside the machine configuration. To keep this material secure, make sure you:
+
+- Encrypt secrets stored in your Talos machine config.
+
+- Limit who has access to machine configuration files, as they contain cluster CA keys and API credentials.
+
+- Rotate certificates regularly, including cluster CA, API server certificates, and kubelet client certificates.
```

Comment:
kubelet client certificates are rotated automatically.

with Omni, there's no good reason to rotate Talos CAs, and Omni currently doesn't support it, so not a good advice either.